nativelib/src/main/resources/gc/immix/Heap.c

```diff
@@ -0,0 +1,253 @@
+#include <stdlib.h>
+#include <sys/mman.h>
+#include <stdio.h>
+#include "Heap.h"
+#include "Block.h"
+#include "Log.h"
+#include "Allocator.h"
+#include "Marker.h"
+#include "State.h"
+#include "utils/MathUtils.h"
+
+#define MAX_SIZE 64 * 1024 * 1024 * 1024L
```

Comment:
We need to do something more sensible here. For example limiting it to the amount of physical memory would be a start.